src/web/messages/transducers.py

```diff
@@ -0,0 +1,320 @@
+    delay: float
+    elementDelays: Optional[List[float]]
+    centerLine: Optional[List[float]]
+    height: Optional[float]
+
+    @classmethod
+    def from_source(
+        cls, source: Union[PhasedArraySource2D, PhasedArraySource3D]
+    ) -> "PhasedArraySettings":
+        """Instantiate the source settings from a source."""
+        if source._focal_length == np.inf:
+            focal_length = None
+        else:
+            focal_length = source._focal_length
+
+        element_delays = None
+        if isinstance(source._configured_element_delays, list):
+            element_delays = source._configured_element_delays
+        elif isinstance(source._configured_element_delays, np.ndarray):
+            element_delays = source._configured_element_delays.tolist()
+
+        return cls(
+            transducerType=TransducerType.phasedArraySource,
+            position=source._position,
+            direction=source._direction,
+            numPoints=source._num_points,
+            numElements=source._num_elements,
+            pitch=source._pitch,
+            elementWidth=source._element_width,
+            tiltAngle=source._tilt_angle,
+            focalLength=focal_length,
+            delay=source._delay,
+            elementDelays=element_delays,
+        )
+
+    def to_ndk_source(self) -> Union[PhasedArraySource2D, PhasedArraySource3D]:
+        """Instantiate the NDK source from transducer settings."""
+        if len(self.position) == 2:
+            return PhasedArraySource2D(
+                position=self.position,
+                direction=self.direction,
+                num_points=self.numPoints,
+                num_elements=self.numElements,
+                pitch=self.pitch,
+                element_width=self.elementWidth,
+                tilt_angle=self.tiltAngle,
+                focal_length=self.focalLength or np.inf,
+                delay=self.delay,
+                element_delays=self.elementDelays,
+            )
+        else:
+            assert self.centerLine is not None
+            assert self.height is not None
+            return PhasedArraySource3D(
+                position=self.position,
+                direction=self.direction,
+                num_points=self.numPoints,
+                num_elements=self.numElements,
+                pitch=self.pitch,
+                element_width=self.elementWidth,
+                tilt_angle=self.tiltAngle,
+                focal_length=self.focalLength or np.inf,
+                delay=self.delay,
+                element_delays=self.elementDelays,
+                center_line=self.centerLine,
+                height=self.height,
+            )
+
+
+class FocusedSourceSettings(_BaseSourceSettings):
+    """Settings for a focused source transducer."""
+
+    transducerType: Literal[TransducerType.focusedSource]
+    position: List[float]
+    aperture: float
+    direction: List[float]
+    focalLength: float
+    numPoints: int
+    delay: float
```

Comment:
The web UI gives some example values for this, which is helpful. Is there also a way to use those example values if I (as a new user) want to just take the default values? Currently, I need to re-type in the values or else the web UI gives me an error:
<img width="536" alt="image" src="https://github.com/agencyenterprise/neurotechdevkit/assets/3221512/1405b660-7934-4fd7-9c10-d1a0d41e6098">

Reply:
Great suggestion! I've implemented it

Reply:
Thanks!